test/evergreen.yml

```diff
@@ -714,7 +714,7 @@ tasks:
     depends_on:
       - name: compile
     commands:
-      - func: "fetch artifacts"
+      - func: "get project"
```

Comment:
Making test tasks depend on the compile task (and fetch artifacts generated by the compile task) is a pattern adopted by wiredtiger Evergreen projects (learned from mongo server Evergreen projects) in order to save the extra compile step (time & resource) in each test task. In the Autoconf/Libtool world, we encountered directory path issues when fetching compile artifacts and running `make check` (or any csuite test) on a different host. And that's why an additional call of `compile wiredtiger` function was appended to ensure a runnable environment before starting test. I guess cmake ran into the same issue. Did the extra `compile wiredtiger` function call (line 718) help?

Reply:
Thanks for the background on why that change was made! 
The extra `compile wiredtiger` step combined with `fetch artifacts` is actually what makes CMake fail (doesn't help). CMake will throw errors if you try to re-configure/re-compile in a build directory that was created on another host. If you change host, a fresh build directory needs to be created. 

Is it ideal we keep `fetch artifacts`? If so, I guess another approach I could take would be updating the configuration function to delete the build directory if it already exists (in the CMake case)?

Reply:
Thought to further understand, In the context of `make-check-test`  do we get any specific resource savings with `fetch artifacts`. Due to the extra compile step, does the fetch artifacts step bring across any artifacts that won't be re-built? i.e I initially thought it be more costly to fetch all the previously built artifacts to only re-compile over them from the extra  `compile wiredtiger` function call (vs just fetching a new clean repo).

Reply:
Thanks for sharing the additional information. You could be right that fetching artifacts is more costly than compile from scratch in this case, as the compile step for wiredtiger is much faster than that of mongo server. I think we can keep the change you made here, and regression-test the task for autoconf.